From a02850c171a04683556e93df62f987deccd0d33c Mon Sep 17 00:00:00 2001
From: Chang Ying <ying.chang@intel.com>
Date: Thu, 23 Nov 2023 10:26:08 +0800
Subject: [PATCH] Change camera serdes power on GPIO polarity.

A1 has changed power on GPIO polarity compared with A0 board.

Tracked-On: OAM-113039
Signed-off-by: Chang Ying <ying.chang@intel.com>
---
 drivers/media/i2c/max96722.c                     | 4 ++--
 drivers/media/platform/intel/ipu6-adlrvp-pdata.c | 1 +
 include/media/max96722.h                         | 1 +
 3 files changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/media/i2c/max96722.c b/drivers/media/i2c/max96722.c
index a75298633c7a..0d4e7a2ac8df 100644
--- a/drivers/media/i2c/max96722.c
+++ b/drivers/media/i2c/max96722.c
@@ -1069,9 +1069,9 @@ static int max96722_poc_enable(struct max96722_priv *priv, bool enable)
 
 		if (info->power_gpio != -1) {
 			if (enable)
-				gpio_set_value(info->power_gpio, 0);
+				gpio_set_value(info->power_gpio, info->power_val);
 			else
-				gpio_set_value(info->power_gpio, 1);
+				gpio_set_value(info->power_gpio, !!!(info->power_val));
 		}
 	}
 
diff --git a/drivers/media/platform/intel/ipu6-adlrvp-pdata.c b/drivers/media/platform/intel/ipu6-adlrvp-pdata.c
index e02e6b1e358b..aeb02e104d70 100644
--- a/drivers/media/platform/intel/ipu6-adlrvp-pdata.c
+++ b/drivers/media/platform/intel/ipu6-adlrvp-pdata.c
@@ -783,6 +783,7 @@ static struct max96722_subdev_info max96722_subdevs[] = {
 		.rx_port = 0,
 		/* GPP_B23	- 0 + 664*/
 		.power_gpio = 687,
+		.power_val = 1,
 		.phy_i2c_addr = 0x42,
 		.alias_addr = 0x43,
 		.ser_type = MAX_SER_96717F,
diff --git a/include/media/max96722.h b/include/media/max96722.h
index 8bebe148a77c..2167cb5f6bbf 100644
--- a/include/media/max96722.h
+++ b/include/media/max96722.h
@@ -29,6 +29,7 @@ struct max96722_subdev_info {
 	struct i2c_board_info board_info;
 	unsigned int rx_port;
 	unsigned int power_gpio;
+	unsigned int power_val;
 	unsigned int phy_i2c_addr;
 	unsigned int alias_addr;
 	unsigned int ser_type;
-- 
2.40.0

